### DISCUSSION TITLE
poly_grain_growth_2D_eldrforce can not use the data from EBSD(TSL)

### DISCUSSION THREAD

**Discussion by zengfy-hust** [idaholab/moose#18615] [Q&A Modules: Solid mechanics] [answered]
Hello Moose:
         I recently made a  test using on the basis of poly_grain_growth_2D_eldrforce.i ,trying to use it to explain my experiment data .
The abnormal grain growth in aluminium alloy .the method I used was shown as following :
         First ,I exported grain file from EBSD(TSL) data and defined the grain number and Euler angle （in radians). 
         second ,I use Excel to change the data into MOOSE format and renamed it as GrainFile.tex  .
         third， I change the input file in poly_grain_growth_2D_eldrforce to GrainFile.tex and reload the module 
however ,the module Stopped due to the error
 *** ERROR ***
The following error occurred in the object "voronoi", of type "PolycrystalVoronoi".

Coupled polycrystal variables differ in periodicity
*******
I dou't know how to fix it ,Any suggestion will be welcome .
Thank you .
in addition ,can ICs of poly_grain_growth_2D_eldrforce change to Reading EBSD data ? can you tell me any module can help explain Abnormal grain growth apart from this module ?
[GrainFile.tex.txt](https://github.com/idaholab/moose/files/6998602/GrainFile.tex.txt)
[poly_grain_growth_2D_eldrforce.i.txt](https://github.com/idaholab/moose/files/6998605/poly_grain_growth_2D_eldrforce.i.txt)

**Comment by zengfy-hust** [comment, marked as answer]
@dschwen

**Reply by GiudGiud** [reply]
@laagesen if one of you has time

**Reply by dschwen** [reply]
All grain order parameters will have to have periodic BCs applied or none. You seem to have a mix (some periodic, some not).

**Reply by zengfy-hust** [reply]
Could you be more specific  about the problem? In fact ,I change the BCs several time ,but the ERROR message kept unchanged .
*** ERROR ***
The following error occurred in the object "voronoi", of type "PolycrystalVoronoi".
Coupled polycrystal variables differ in periodicity
********
How could I  change the voroni or BCs to make them correspond to each other ? Thank you so much!@dschwen
